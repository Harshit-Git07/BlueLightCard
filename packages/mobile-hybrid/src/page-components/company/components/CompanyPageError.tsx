import InvokeNativeLifecycle from '@/invoke/lifecycle';
import { Button, Heading } from '@bluelightcard/shared-ui';
import React, { FC } from 'react';

const lifecycleEvent = new InvokeNativeLifecycle();

type props = {
  message: string;
};

const CompanyPageError: FC<props> = ({ message }) => {
  return (
    <>
      <Heading headingLevel={'h2'} className="dark:text-white text-black pt-8">
        {message}
      </Heading>
      <Button type="link" href="/" onClick={() => lifecycleEvent.lifecycleEvent('onBackPressed')}>
        Return Home
      </Button>
    </>
  );
};

export default CompanyPageError;
