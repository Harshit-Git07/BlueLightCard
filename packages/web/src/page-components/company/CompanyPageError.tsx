import Heading from '@bluelightcard/shared-ui/components/Heading';
import { Button } from '@bluelightcard/shared-ui/index';
import React, { FC } from 'react';

type props = {
  message: string;
};

const CompanyPageError: FC<props> = ({ message }) => {
  return (
    <div className="w-full">
      <Heading headingLevel={'h2'} className="dark:text-white text-black pt-8">
        {message}
      </Heading>
      <Button href="/">Return Home</Button>
    </div>
  );
};

export default CompanyPageError;
