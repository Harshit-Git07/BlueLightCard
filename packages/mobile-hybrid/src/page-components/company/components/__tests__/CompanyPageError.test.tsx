import { render } from '@testing-library/react';
import CompanyPageError from '../CompanyPageError';

describe('Company Page Error Component', () => {
  it('should render correctly', () => {
    const { container } = render(<CompanyPageError message="Failed to load" />);
    expect(container).toMatchSnapshot();
  });
});
