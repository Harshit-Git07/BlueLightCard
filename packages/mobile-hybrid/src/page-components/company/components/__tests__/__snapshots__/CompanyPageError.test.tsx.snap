// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Company Page Error Component should render correctly 1`] = `
<div>
  <h2
    class="text-2xl text-heading-colour dark:text-heading-colour-dark mb-2 font-typography-title-large font-typography-title-large-weight text-typography-title-large leading-typography-title-large tracking-typography-title-large text-colour-onSurface-light dark:text-colour-onSurface-dark
    tablet:font-typography-headline-weight tablet:font-typography-headline tablet:text-typography-headline tablet:tracking-typography-headline tablet:leading-typography-headline dark:text-white text-black pt-8"
  >
    Failed to load
  </h2>
  <a
    class="inline-block py-1.5 px-5 rounded-md transition border-2 focus:outline outline-2 outline-offset-2  text-button-primary-default-label-colour-light dark:text-button-primary-default-label-colour-dark hover:bg-button-primary-hover-bg-colour-light dark:hover:bg-button-primary-hover-bg-colour-dark hover:text-button-primary-hover-label-colour-light dark:hover:text-button-primary-hover-label-colour-dark focus:outline-[#2EB8E6] dark:focus:outline-[#FFFF00] bg-button-primary-default-bg-colour-light dark:bg-button-primary-default-bg-colour-dark border-transparent , text-button-label-font, font-button-label-font font-button-label-font-weight tracking-button-label-font leading-button-label-font"
    href="/"
  >
    Return Home
  </a>
</div>
`;
